distance-metrics/benches/all.rs

```diff
@@ -0,0 +1,44 @@
+use bencher::{benchmark_group, benchmark_main, Bencher};
+
+use distance_metrics::{
+    metric::Metric,
+    simple::{CosineMetric, EuclidMetric},
+};
+use rand::{rngs::StdRng, Rng, SeedableRng};
+
+benchmark_main!(benches);
+benchmark_group!(
+    benches,
+    legacy,
+    non_simd,
+    simple::<EuclidMetric>,
+    simple::<CosineMetric>
+);
+
+fn legacy(bench: &mut Bencher) {
```

Comment:
Would be curious to see some benchmarking results in this PR for the legacy approach vs your new implementations!

Reply:
The most recent results:
```
test legacy               ... bench:          21 ns/iter (+/- 0)
test non_simd             ... bench:         183 ns/iter (+/- 2)
test simple<CosineMetric> ... bench:          22 ns/iter (+/- 0)
test simple<EuclidMetric> ... bench:          25 ns/iter (+/- 0)
```

Reply:
Yes, the `legacy` is euclid but with square root. I believe we don't care about having squared values. If I drop the square root calculation then they are almost the same (2-5% difference):

```
test legacy               ... bench:          54 ns/iter (+/- 6)
test metric<CosineMetric> ... bench:          56 ns/iter (+/- 11)
test metric<EuclidMetric> ... bench:          55 ns/iter (+/- 6)
test non_simd             ... bench:         441 ns/iter (+/- 40)
```
(the numbers are higher because I'm on battery and the CPU is throttled)

We could use that since it doesn't have dimensions limitation. Another idea would be to have the current metric (as for example `Euclid300`) for the use case of 300 dimensions.